third-party/cmake/BuildLuajit.cmake

```diff
@@ -51,30 +52,58 @@ if(CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
 else()
   set(AMD64_ABI "")
 endif()
-set(INSTALLCMD_UNIX ${MAKE_PRG} CFLAGS=-fPIC
-                                CFLAGS+=-DLUA_USE_APICHECK
-                                CFLAGS+=-funwind-tables
-                                ${NO_STACK_CHECK}
-                                ${AMD64_ABI}
-                                CCDEBUG+=-g
-                                Q=
-                                install)
+set(BUILDCMD_UNIX ${MAKE_PRG} CFLAGS=-fPIC
+                              CFLAGS+=-DLUA_USE_APICHECK
+                              CFLAGS+=-funwind-tables
+                              ${NO_STACK_CHECK}
+                              ${AMD64_ABI}
+                              CCDEBUG+=-g
+                              Q=)
 
-if(UNIX)
-  if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-    if(CMAKE_OSX_DEPLOYMENT_TARGET)
-      set(DEPLOYMENT_TARGET "MACOSX_DEPLOYMENT_TARGET=${CMAKE_OSX_DEPLOYMENT_TARGET}")
-    else()
-      # Use the same target as our nightly builds
-      set(DEPLOYMENT_TARGET "MACOSX_DEPLOYMENT_TARGET=10.11")
-    endif()
+if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+  if(CMAKE_OSX_DEPLOYMENT_TARGET)
+    set(DEPLOYMENT_TARGET "MACOSX_DEPLOYMENT_TARGET=${CMAKE_OSX_DEPLOYMENT_TARGET}")
   else()
-    set(DEPLOYMENT_TARGET "")
+    # Use the same target as our nightly builds
+    set(DEPLOYMENT_TARGET "MACOSX_DEPLOYMENT_TARGET=10.11")
   endif()
+else()
+  set(DEPLOYMENT_TARGET "")
+endif()
 
-  BuildLuaJit(INSTALL_COMMAND ${INSTALLCMD_UNIX}
+if(UNIX AND NOT CMAKE_OSX_ARCHITECTURES)
+  BuildLuaJit(INSTALL_COMMAND ${BUILDCMD_UNIX}
     CC=${DEPS_C_COMPILER} PREFIX=${DEPS_INSTALL_DIR}
-    ${DEPLOYMENT_TARGET})
+    ${DEPLOYMENT_TARGET} install)
+
+elseif(CMAKE_OSX_ARCHITECTURES AND APPLE)
+
+  set(LUAJIT_SRC_DIR "${DEPS_BUILD_DIR}/src/luajit")
+  foreach(ARCH IN LISTS CMAKE_OSX_ARCHITECTURES)
+    set(STATIC_CC "${LUAJIT_C_COMPILER} -arch ${ARCH}")
+    set(DYNAMIC_CC "${LUAJIT_C_COMPILER} -arch ${ARCH} -fPIC")
+    set(TARGET_LD "${LUAJIT_C_COMPILER} -arch ${ARCH}")
+    list(APPEND LUAJIT_THIN_EXECUTABLES "${LUAJIT_SRC_DIR}-${ARCH}/src/luajit")
+    list(APPEND LUAJIT_THIN_STATIC_LIBS "${LUAJIT_SRC_DIR}-${ARCH}/src/libluajit.a")
+    list(APPEND LUAJIT_THIN_DYLIBS "${LUAJIT_SRC_DIR}-${ARCH}/src/libluajit.so")
+    list(APPEND LUAJIT_THIN_DEPS "luajit-${ARCH}")
+
+    BuildLuaJit(TARGET "luajit-${ARCH}"
+        BUILD_COMMAND ${BUILDCMD_UNIX}
+        CC=${LUAJIT_C_COMPILER} STATIC_CC=${STATIC_CC}
+        DYNAMIC_CC=${DYNAMIC_CC} TARGET_LD=${TARGET_LD}
+        PREFIX=${DEPS_INSTALL_DIR}
+        ${DEPLOYMENT_TARGET})
```

Comment:
For reference (I should/will add this to a commit message):

The extra flags are taken from https://luajit.org/install.html, in the section on cross-compiling.

We could technically define a `CROSS` variable instead, but this would require creating executable stubs like `arm64-apple-darwin-clang` that basically just call `clang -arch arm64`.